File: P2.py

## Problem 2
# Single number (https://leetcode.com/problems/single-number/)
# TC : O(n)
# SC :O(1)
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # if len(nums)==0:
        #     return 
        if len(nums)==1:
            return nums[-1]
        p = nums[0] ^ nums[1]
        if len(nums)<=2:
            return p
        for i in range(2, len(nums)):
            p = p ^ nums[i] 
        return p

File: p2.java
// Time Complexity : O(n)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no


// Your code here along with comments explaining your approach

class Solution {
    public int singleNumber(int[] nums) {
        int result = 0;
        //Do xor of all numbers, double numbers cancels out
        for(int i=0; i<nums.length; i++){
            result ^= nums[i]; 
        }
        return result;
    }
}